client/state/data-layer/wpcom/comments/index.js

```diff
@@ -160,35 +159,66 @@ export const announceFailure = ( { dispatch, getState }, { siteId, postId } ) =>
 	dispatch( errorNotice( error ) );
 };
 
-export const deleteComment = ( { dispatch }, action ) => {
-	const { siteId, commentId } = action;
+// @see https://developer.wordpress.com/docs/api/1.1/post/sites/%24site/comments/%24comment_ID/delete/
+export const deleteComment = ( { dispatch, getState }, action ) => {
+	const { siteId, postId, commentId } = action;
+	const comment = getSiteComment( getState(), siteId, commentId );
+
+	dispatch( removeComment( siteId, postId, commentId ) );
+
 	dispatch(
-		http( {
-			method: 'POST',
-			apiVersion: '1.1',
-			path: `/sites/${ siteId }/comments/${ commentId }/delete`,
-			onSuccess: {
-				...action,
-				type: COMMENTS_DELETE_SUCCESS,
+		http(
+			{
+				method: 'POST',
+				apiVersion: '1.1',
+				path: `/sites/${ siteId }/comments/${ commentId }/delete`,
 			},
-			onFailure: {
+			{
 				...action,
-				type: COMMENTS_DELETE_FAILURE,
-			},
-		} )
+				comment,
+			}
+		)
 	);
 };
 
-export const removeCommentFromState = ( { dispatch }, action ) => {
-	const { siteId, postId, commentId } = action;
-	dispatch( removeComment( siteId, postId, commentId ) );
+export const announceDeleteSuccess = ( { dispatch } ) => {
+	dispatch(
+		createNotice(
+			'is-error',
+			translate( 'Comment deleted permanently.' ),
+			{
+				duration: 5000,
+				isPersistent: true,
+			}
+		)
+	);
 };
 
-export const announceDeleteFailure = ( { dispatch } ) => {
-	dispatch( errorNotice( translate( 'Could not delete the comment' ) ) );
+export const announceDeleteFailure = ( { dispatch, getState }, action ) => {
+	const { siteId, postId, comment } = action;
+
+	dispatch(
+		errorNotice(
+			translate( 'Could not delete the comment.' ),
+			{
+				duration: 5000,
+				isPersistent: true,
+			}
+		)
+	);
+
+	if ( comment ) {
+		dispatch( {
+			type: COMMENTS_RECEIVE,
+			siteId,
+			postId,
+			comments: [ comment ],
+			skipSort: !! get( comment, 'parent.ID' ),
+		} );
+	}
 };
 
 export default {
 	[ COMMENTS_REQUEST ]: [ dispatchRequest( fetchPostComments, addComments, announceFailure ) ],
-	[ COMMENTS_DELETE ]: [ dispatchRequest( deleteComment, removeCommentFromState, announceDeleteFailure ) ],
+	[ COMMENTS_DELETE ]: [ dispatchRequest( deleteComment, announceDeleteSuccess, announceDeleteFailure ) ],
```

Comment:
I'm still confused on `COMMENTS_REMOVE` vs `COMMENTS_DELETE`? We shouldn't need to dispatch an action from here in order that the reducers actually remove the comment from state. Is there a difference between the semantics of `_DELETE` and `_REMOVE`?